### ISSUE TITLE
Helm charts should include `airbyte-proxy` to secure API and webapp

### ISSUE THREAD

**Issue by evantahler** [airbytehq/airbyte#17866]

**Comment by evantahler** [comment]
cc @nicor88

**Comment by supertopher** [comment]
@davinchia I don't think I understand the upsides to this change.  On some level I understand having a layer of basic auth.  More context on what we want to get out of this would help me prioritize ProdEng's effort here

**Comment by evantahler** [comment]
I can chime in!

We want to provide OSS Airbyte users with a layer of auth by default.  Yes, we tell folks they are responsible for securing their server, but folks will always [forget](https://airbyte.com/blog/improving-security-for-open-source-airbyte-users), and so we've adopted a ["secure by default"](https://docs.google.com/document/d/1oETYBlih3CIP9l89Ht56elXAPMc74MrTJnQT1bxOv0A/edit#heading=h.dvhr62h2hp70) product principal going forward.

The long-term goal is to add users and APIKeys to OSS Airbyte, just like we have for cloud.  In the short term, I threw up this Nginx proxy to do... something.  Auth is configured via environment variables now, and this works for docker-compose OSS users, but not K8s users.  This story is provide the same experience for K8s users (although you don't have to use the same tools)
* By default, the Airbyte webapp and API are protected behind basic auth
* Users can easily configure the name & password by which they authenticate

**Comment by davinchia** [comment]
Thanks Evan!

@supertopher the simplest way to explain this is to try and run Airbyte Docker locally. You've noticed a password page. We want to enable the same flow in Airbyte Kubernetes.

In addition to this, @benmoriceau is hoping to also leverage the Airbyte Proxy container to blue/green deploy controller changes as we convert the server over to micronaut. He's blocked by the lack of this in Kube for now.

Doing so will:
1) Bring Airbyte Kube security inline with Airbyte Docker
2) Allow the old/new controller implementations to live side by side and let Benoit merge in code piecemeal instead of having a long running feature branch as he slowly converts routes.

**Comment by fabianofpena** [comment]
This is really missing! I'm using airbyte on Kubernetes due to scalability. But the security is poor.

**Comment by czomo** [comment]
Hi @evantahler ! I've created draft for airyte-proxy. I think there is one issue with current implementation. As everything resides on kubernetes the service URL are different that in docker-compose. Can we make  PROXY_PASS_WEB and PROXY_PASS_API configurable?

https://github.com/airbytehq/airbyte-platform/pull/228

**Comment by evantahler** [comment]
I'm probably not the correct person to take a look at this, so I've asked @davinchia and @marcosmarxm to review your PR!

**Comment by octavia-squidington-iii** [comment]
At Airbyte, we seek to be clear about the project priorities and roadmap.  This issue has not had any activity for 180 days, suggesting that it's not as critical as others.  It's possible it has already been fixed. It is being marked as stale and will be closed in 20 days if there is no activity. To keep it open, please comment to let us know why it is important to you and if it is still reproducible on recent versions of Airbyte.

**Comment by marcosmarxm** [comment]
Closing as `airbyte-proxy` won't be add to Helm deployment instead it will be add a Keycloack layer.

**Comment by dantonbertuol** [comment]
> Closing as `airbyte-proxy` won't be add to Helm deployment instead it will be add a Keycloack layer.

Hi @marcosmarxm, Any predictions for this implementation? It is an issue that the community has expressed a lot of interest in.

**Comment by marcosmarxm** [comment]
Yes, this feature is almost ready. It will likely be included in upcoming releases. Stay updated.

**Comment by dantonbertuol** [comment]
> Yes, this feature is almost ready. It will likely be included in upcoming releases. Stay updated.

@marcosmarxm, just one more thing, will it work on the community version?

**Comment by dantonbertuol** [comment]
> Yes, this feature is almost ready. It will likely be included in upcoming releases. Stay updated.

Hi @marcosmarxm, any news about it?
